doc/neps/nep-0019-rng-policy.rst

```diff
@@ -0,0 +1,208 @@
+Blocked by our policy, and our inability to work around that policy, many of
+those contributors simply walked away.
+
+
+Implementation
+--------------
+
+We propose first freezing ``RandomState`` as it is and developing a new RNG
+subsystem alongside it.  This allows anyone who has been relying on our old
+stream-compatibility guarantee to have plenty of time to migrate.
+``RandomState`` will be considered deprecated, but with a long deprecation
+cycle, at least a few years.  Deprecation warnings will start silent but become
+increasingly noisy over time.  Bugs in the current state of the code will *not*
+be fixed if fixing them would impact the stream.  However, if changes in the
+rest of ``numpy`` would break something in the ``RandomState`` code, we will
+fix ``RandomState`` to continue working (for example, some change in the
+C API).  No new features will be added to ``RandomState``.  Users should
+migrate to the new subsystem as they are able to.
+
+Work on a proposed `new PRNG subsystem
+<https://github.com/bashtage/randomgen>`_ is already underway.  The specifics
+of the new design are out of scope for this NEP and up for much discussion, but
+we will discuss general policies that will guide the evolution of whatever code
+is adopted.
+
+First, we will maintain API source compatibility just as we do with the rest of
+``numpy``.  If we *must* make a breaking change, we will only do so with an
+appropriate deprecation period and warnings.
+
+Second, breaking stream-compatibility in order to introduce new features or
+improve performance will be *allowed* with *caution*.  Such changes will be
+considered features, and as such will be no faster than the standard release
+cadence of features (i.e. on ``X.Y`` releases, never ``X.Y.Z``).  Slowness is
+not a bug.  Correctness bug fixes that break stream-compatibility can happen on
+bugfix releases, per usual, but developers should consider if they can wait
+until the next feature release.  We encourage developers to strongly weight
+user’s pain from the break in stream-compatibility against the improvements.
+One example of a worthwhile improvement would be to change algorithms for
+a significant increase in performance, for example, moving from the `Box-Muller
+transform <https://en.wikipedia.org/wiki/Box%E2%80%93Muller_transform>`_ method
+of Gaussian variate generation to the faster `Ziggurat algorithm
+<https://en.wikipedia.org/wiki/Ziggurat_algorithm>`_.  An example of an
+unworthy improvement would be tweaking the Ziggurat tables just a little bit.
+
+Any new design for the RNG subsystem will provide a choice of different core
+uniform PRNG algorithms.  We will be more strict about a select subset of
+methods on these core PRNG objects.  They MUST guarantee stream-compatibility
+for a minimal, specified set of methods which are chosen to make it easier to
+compose them to build other distributions.  Namely,
+
+    * ``.bytes()``
+    * ``.random_uintegers()``
```

Comment:
One of these (bytes, uintegers) seems redundant.  uintegers should probably by 64 bit.